libs/payments/types/src/lib/invoice.factory.ts

```diff
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { faker } from '@faker-js/faker';
+import { Stripe } from 'stripe';
+
+export const InvoiceFactory = (
+  override: Partial<Stripe.Invoice>
+): Stripe.Invoice => ({
+  id: `in_${faker.string.alphanumeric({ length: 24 })}`,
+  object: 'invoice',
+  account_country: null,
+  account_name: null,
+  account_tax_ids: null,
+  amount_due: 0,
+  amount_paid: 0,
+  amount_remaining: 0,
+  amount_shipping: 0,
+  application: null,
+  application_fee_amount: null,
+  attempt_count: 1,
+  attempted: true,
+  automatic_tax: {
+    enabled: true,
+    status: 'complete',
+  },
+  billing_reason: null,
+  charge: faker.string.alphanumeric(10),
+  collection_method: 'charge_automatically',
+  created: 1,
+  currency: 'USD',
+  custom_fields: null,
+  customer: `cus_${faker.string.alphanumeric({ length: 14 })}`,
+  customer_address: null,
+  customer_email: faker.internet.email(),
+  customer_name: null,
+  customer_phone: null,
+  customer_shipping: null,
+  customer_tax_exempt: null,
+  default_payment_method: faker.string.alphanumeric(10),
+  default_source: faker.string.alphanumeric(10),
+  default_tax_rates: [],
+  description: null,
+  discount: null,
+  discounts: null,
+  due_date: null,
+  ending_balance: null,
+  footer: null,
+  from_invoice: null,
+  last_finalization_error: null,
+  latest_revision: null,
+  lines: {
+    object: 'list',
+    data: [
+      {
+        id: faker.string.alphanumeric(10),
+        object: 'line_item',
+        amount: 500,
+        amount_excluding_tax: 500,
+        currency: 'USD',
+        description: null,
+        discount_amounts: null,
+        discountable: true,
+        discounts: null,
+        livemode: false,
+        metadata: {},
+        period: {
+          end: 1,
+          start: 0,
+        },
+        plan: null,
+        price: null,
+        proration: false,
+        proration_details: null,
+        quantity: null,
+        subscription: null,
+        type: 'subscription',
+        unit_amount_excluding_tax: null,
```

Comment:
It's really hard to work with sub-arrays like this one if you want to override specific information inside, since it nukes the whole array. Instead, we should break these sub-arrays into their own factories, and this current factory should use that factory to build it's array of items.

Reply:
Took a pass at what you're requesting; if sufficient, go ahead and mark this as resolved, if not, just let me know.